Add support for getting module eeprom

mickep76 opened the pull request:
- Added support for getting module eeprom like SFP's, allows for getting Transceiver Vendor, Product, S/N
- Added an example

mickep76 commented:
Same as "ethtool -m eth0 hex on"

mickep76 commented:
Started adding module info

mickep76 commented:
That finished support for module eeprom and SFF-8709 info.

safchain commented on ethtool.go:
```diff
@@ -117,6 +140,42 @@ func (e *Ethtool) BusInfo(intf string) (string, error) {
 	return string(bytes.Trim(info.bus_info[:], "\x00")), nil
 }
 
+func (e *Ethtool) ModuleEeprom(intf string) (string, error) {
```
Could you add the documentation for each exported function

safchain commented on sff_8436.go:
```diff
@@ -0,0 +1,9 @@
+package ethtool
```
could you add the license header for each new file

safchain commented on example/main.go:
```diff
@@ -0,0 +1,72 @@
+	"fmt"
+	"log"
+
+	//	"github.com/safchain/ethtool"
```
correct package has to used, probably debug stuff remaining :)

safchain commented on ethtool.go:
```diff
@@ -138,6 +197,44 @@ func (e *Ethtool) getDriverInfo(intf string) (ethtoolDrvInfo, error) {
+	}
+
+	if modInfo.eeprom_len > EEPROM_LEN {
+		return ethtoolEeprom{}, ethtoolModInfo{}, fmt.Errorf("eeprom size: %d is larger then buffer size: %d", modInfo.eeprom_len, EEPROM_LEN)
```
nit: s/then/than/

mickep76 commented:
I'm adding additional code for QFP 8436, so better to merge until after that. I'll address your comments as-well.

mickep76 commented:
I will split out the SFF decode part in a separate package otherwise you'll have to deal with license conflict GPL from orig. ethtool vs Apache. Since the code is basically a copy/paste of their code from C to Go. I have to respect their original license. https://github.com/mickep76/sff-eeprom

mickep76 commented:
Fixed the comments in the review, sry for the delay. I implemented a module decoder separately mostly because it needed a diff. license. I ended up completely re-factoring anyway so I can merge it in to this code base if you think that make sense? (https://github.com/mickep76/go-sff)
